### PR TITLE
JCRVLT-557 - make default for overwriting node types configurable (OSGi)

### PR DESCRIPTION
**Pull request by reschke**

### CODE REVIEW

File: vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/FolderArtifactHandler.java [hunk 1]
```diff
@@ -73,6 +78,18 @@ public void setNodeType(String nodeType) {
         this.nodeType = nodeType;
     }
 
+    /**
+     * Sets whether primary type of folders should be overwritten
+     * 
+     * @param primary
```
Comment (kwin) [inline vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/FolderArtifactHandler.java:84]: wrong param and confusing description

**Review by kwin — APPROVED** [review]
LGTM except for the minor issue with the javadoc mentioned above and the 4 errors during javadoc creation (https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Ffilevault/detail/PR-175/1/pipeline). Please squash before merging.

**Comment by reschke** [comment]
Thanks, will do.

**Comment by reschke** [comment]
@kwin - two more questions:

1. Is it ok that I *removed* (not deprecated) " public FSPackageRegistry(@NotNull File homeDir, InstallationScope scope, @Nullable AbstractPackageRegistry.SecurityConfig securityConfig, boolean isStrict)"
2. Is there work left to do in "PackageTaskOptionsSerializer"?

**Comment by kwin** [comment]
> 1. Is it ok that I _removed_ (not deprecated) " public FSPackageRegistry(@NotNull File homeDir, InstallationScope scope, @nullable AbstractPackageRegistry.SecurityConfig securityConfig, boolean isStrict)"

Yes, that is an impl class and should not be used anywhere outside FileVault (although this is only enforced with OSGi)

> 2. Is there work left to do in "PackageTaskOptionsSerializer"?
Good point, although only used in https://github.com/apache/sling-org-apache-sling-feature-extension-content/blob/7f982174eb877a3a0c7bd6b69862a3a11e9db137/src/main/java/org/apache/sling/feature/extension/content/ContentHandler.java#L82 AFAIK it would be good to serialize all options. We should have some more test coverage for this class as well.

**Comment by reschke** [comment]
@kwin - PTAL at https://github.com/apache/jackrabbit-filevault/pull/175/commits/df384148c13b2550ec6e26323144c49fedb19e1c -- one puzzling thing here is that "PackageTaskOptionsSerializer" requires a method in "ImportOptions" that is deprecated; we should either undeprecate it, or see what needs to be changes in the serializer.

**Comment by kwin** [comment]
@reschke Thanks for the update. LGTM, feel free to squash and merge. I am gonna take care of the deprecated method usage in ImportOptions.

**Pull request merged**